### ISSUE TITLE
NC 28.0.x - External Folders are not scanned for new media.

### ISSUE THREAD

**Issue by Moltey** [nextcloud/photos#2394]
**Describe the bug**
Since upgrading to NC 28, my external drives (permanently mounted folders from the file system) are no longer scanned regularly for new media. The photos are accessible via normal folder view.

**To Reproduce**
1. Mount external folder.
2. Add new media.
3. Wait.
4. The new media will not shown in the photos overview page.

**Expected behavior**
4. New media should show up in the photos overview page.

**Comment by claushviid** [comment]
i have exactly the same problem

**Comment by Moltey** [comment]
I just wanted to add, that I am scanning external media each night with following command:

```bash
$ podman exec --user 9999 -t nextcloud-cron ./occ files:scan bob --path=/bob/files/MyPhotos
```

...whereas the user ID is aligned with my setup, which was working perfectly fine before NC 28.

**Comment by Beer17HWAM** [comment]
When can we expect this issue is fixed?

**Comment by ag-ubrio** [comment]
I have the same problem on NC 29. 
The external storage (SFTP connection) is not shown nowhere in the Photo App, it is even not possible to add photos from there manualy to albums.
Please ask if any additional info is needed or useful to fix this.

**Comment by A37stKpDodi4BEr65TXb** [comment]
Have similar issue since upgrading to 29.0.4 from 28. 
My photos are arranged in external folders /Photos/YEAR/MONTH. And /Photos is set as media folder under settings.

Currently photo app only displays media files dated July 2017 and older (except in Folders tab), and refuses to show anything more recent.

**Comment by Moltey** [comment]
> Are the mount locations of the external folders listed as Media Folders under _Photos settings_?

Yes, the folder is correctly set up in the Photos settings (menu opens after clicking the settings icon in the bottom left).

Please note: After each patch of NC, the update mechanism seems to trigger some kind of script that allows NC to detect photos that have been added since the last upgrade. However, this is always a one-time event after an upgrade. Media added after the upgrade will not show up.

**Comment by ag-ubrio** [comment]
> Are the mount locations of the external folders listed as Media Folders under _Photos settings_?

Yes, correctly set.
But your question made me playing around with the Media Folders and I realised that the problem only occours if the "root" folder of the external storage is set as media folder (e.g. I have set the "SofortUpload"-folder to be external. When I set "SofortUpload" as Media Folder, the images in the folder are not shown). 

But when I set a folder below this folder, e.g. the subfolder "Camera", the fotos in the subfolder are imediately shown. They apear so quickly that I presume that the preview-images must have been created before, the fotos just where not shown.

**Comment by ruedigerkupper** [comment]
I confirm this behavior (NC 29):
- If in your settings the media folder is set to "All Folders", images from external folders do NOT show up in the "All Media" view. (But they can be seen in the "Folders" view.
- If you explicitly add the external folder in settings -> media folder, images show up in the "All Media" view.

**Comment by ruedigerkupper** [comment]
See also #575. Groupfolders are not shown in "All Media" view by design, due to performance concerns(?). Possibly the same decision was made regrding external storage?

**Comment by Moltey** [comment]
> I confirm this behavior (NC 29):
> 
>     * If in your settings the media folder is set to "All Folders", images from external folders do NOT show up in the "All Media" view. (But they can be seen in the "Folders" view.)
> 
>     * If you explicitly add the external folder in settings -> media folder, images DO show up in the "All Media" view.

To show media, this is also required in my setup. I thought it was a feature. 

However, the trick does **not** work for me: 
**Newly added media** is still not shown in the media/picture overview. I can only browse newly added media via the folder button on the left menu (as explained in my very first post).

**Comment by ruedigerkupper** [comment]
It wasn't meant to be a solution to your original problem, I just confirmed @ag-ubrio 's findings.
But you are right, these are two separate issues, though possibly related. I will file a separate bug report for it.

**Comment by Beer17HWAM** [comment]
Still not working. No thumbnails from external storage on Maps.
